lib/ami_spec.rb

```diff
@@ -57,6 +58,11 @@ def self.run(options)
       options[:key_file] = key_pair.key_file
     end
 
+    if options[:aws_security_groups].nil? || options[:aws_security_groups].empty?
+      security_group = AwsSecurityGroup.create(ec2: ec2, subnet_id: options[:subnet_id], logger: logger)
```

Comment:
```suggestion
      temporary_security_group = AwsSecurityGroup.create(ec2: ec2, subnet_id: options[:subnet_id], logger: logger)
```

Just to make it really clear this _only_ stores the temp group. I panicked a little when I saw it was deleting the security group, then realised this only stores the temporary group if it exist